data/abilities.js

```diff
@@ -1449,9 +1449,10 @@ exports.BattleAbilities = {
 		onEnd: function (pokemon) {
 			if (pokemon.illusion) {
 				this.debug('illusion cleared');
+				let illusion = pokemon.illusion;
 				pokemon.illusion = null;
 				let details = pokemon.template.species + (pokemon.level === 100 ? '' : ', L' + pokemon.level) + (pokemon.gender === '' ? '' : ', ' + pokemon.gender) + (pokemon.set.shiny ? ', shiny' : '');
-				this.add('replace', pokemon, details);
+				this.add('replace', pokemon, details, '[illusion] ' + illusion);
```

Comment:
You don't need to do this, you know. You can just read the Pokémon's searchid pre-replacement.

Reply:
Does the pre-replacement searchid still exist during this step on the client? Because I could not find anywhere that this would be stored, and it isn't in this message before...

Reply:
Never mind I figured this out.